File: python/leetcode205.py
class Solution(object):
    def isIsomorphic(self, s, t):
        """
        :type s: str
        :type t: str
        :rtype: bool
        """
        ds = {}
        dt = {}
        if len(s) != len(t):
            return False
        for i in range(len(s)):
            if s[i] not in ds and t[i] not in dt:
                ds[s[i]] = t[i]
                dt[t[i]] = s[i]
            elif s[i] in ds and t[i] in dt:
                if ds[s[i]] == t[i] and  dt[t[i]] == s[i]:
                    continue
                else:
                    return False
            else:
                return False
        return True

File: leetcodec++/leetcode205.cpp
//
// Created by 黄敏辉 on 2019-08-30.
//

class Solution {
public:
    bool isIsomorphic(string s, string t) {
        unordered_map<char, char> s_t;
        unordered_map<char, char> t_s;

        int l1 = s.size();
        int l2 = t.size();

        if (l1 != l2)
            return false;

        for(int i = 0; i < l1; i++){
            if (s_t.find(s[i]) == s_t.end() && t_s.find(t[i]) == t_s.end()){
                s_t[s[i]] = t[i];
                t_s[t[i]] = s[i];
            } else if ( (s_t.find(s[i]) == s_t.end() && t_s.find(t[i]) != t_s.end()) || (s_t.find(s[i]) != s_t.end() && t_s.find(t[i]) == t_s.end()))
                return false;
            else if (s_t[s[i]] != t[i] || t_s[t[i]] != s[i])
                return false;
        }
        return true;
    }
};